numpy/lib/stride_tricks.py

```diff
@@ -111,6 +112,190 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
     return view
 
 
+def _sliding_window_view_dispatcher(x, window_shape, axis=None, *,
+                                    subok=None, writeable=None):
+    return (x,)
+
+
+@array_function_dispatch(_sliding_window_view_dispatcher, module='numpy')
+def sliding_window_view(x, window_shape, axis=None, *, subok=False, writeable=False):
+    """
+    Create a sliding window view into the array with the given window shape.
+
+    Creates a sliding window view of the N dimensional array with the given
+    window shape. Window slides across each dimension of the array and extract
+    a subsets of the array at any window position.
+
+    Parameters
+    ----------
+    x : array_like
+        Array to create the sliding window view from.
+
+    window_shape : int or tuple of int
+        The shape of the window. If `axis` is not present, must have same
+        length as the number of input array dimensions.
+        Single integers `i` are treated as if they were the tuple `(i,)`.
+
+    axis : int or tuple of int, optional
+        If present, `window_shape[i]` will refer to the axis `axis[i]` of `x`,
+        otherwise `window_shape[i]` will refer to axis `i` of `x`.
+        Single integers `i` are treated as if they were the tuple `(i,)`.
+
+    subok : bool, optional
+        If True, sub-classes will be passed-through, otherwise the returned
+        array will be forced to be a base-class array (default).
+
+    writeable : bool, optional
+        When true, allow writing to the returned view. The default is false,
+        as this should be used with caution: the returned view contains the
+        same memory location multiple times, so writing to one location will
+        cause others to change.
+
+    Returns
+    -------
+    view : ndarray
+        Sliding window view of the array. The sliding window dimensions are
+        inserted at the end, and the original dimensions are trimmed as
+        required by the size of the sliding window. That is,
+        `view.shape = x_shape_trimmed + window_shape`, where `x_shape_trimmed` is
+        `x.shape` with every entry reduced by the size of the window in the
+        respective axis minus one.
```

Comment:
Can you try just giving the formula here? From reading the text, it isn't clear whether "minus one" refers to the size of the window being subtracted or the final result. E.g., is it "input_size - (window_size - 1)" or "input_size - window_size - 1"?

Reply:
The former.

Reply:
Clarified docstring.